File: 0258-add-digits/0258-add-digits.py
import math
class Solution:
    def addDigits(self, num: int) -> int: 
        total = 0
        if num < 10:
            return num
        
        while num > 9:  
            
            while num > 0:
                total += (num % 10)
                num //= 10
                
            if(total < 10):
                break
                
            num = total
            total = 0
            
        return total

File: 0258-add-digits/0258-add-digits.cpp
class Solution {
public:
    int addDigits(int num) 
    {
        if(num < 10)
            return num;
            

        int result = 0;
        
        while(num > 9)
        {
            while(num > 0 )
            {  
                result += num % 10;
                
                num = num / 10;
            }
            
            if (result < 10)
                break;
            
            num = result;
            result = 0;
        }
        
        return result;
    }
};